zulipterminal/core.py

```diff
@@ -531,8 +531,14 @@ def extract_keywords_from_search_query(
     def search_messages(self, text: str) -> None:
         # Search for a text in messages
         self.model.index["search"].clear()
-        self.model.set_search_narrow(text)
-
+        modified_search_text = (
+            text + self.model.addtional_search_text_for_current_narrow()
+        )
+        text_without_keyword, search_narrow = self.extract_keywords_from_search_query(
+            modified_search_text
+        )
+        self.model.narrow = search_narrow
```

Comment:
My primary remaining concern, given we're potentially taking a pause on this after your work, is this direct setting of the narrow here. Doing so means we potentially don't update other data which would otherwise be set, via eg. model.set_narrow or _narrow_to or something which also updated the UI.

Right now the positive is that the narrow *shouldn't* change, unless a user enters extra search text - and not all combinations work. For now I'd suggest we alert the user with a simple warning popup (popup_with_message), if the search looks like a complicated search which would require setting the narrow, ie. if the old and new narrows wouldn't match.

Examples of cases where this can happen is being in all messages, and adding eg. `is:starred some search text`, in which case the narrow changes. That 'works', but then adding `stream:general` or similar does not. So it'd be good to be consistent for now.